test/test_metapackage.py

```diff
@@ -76,6 +86,16 @@ def test_validate_metapackage(self):
                 else:
                     _validate_metapackage(path, package)
 
+    def test_collect_warnings(self):
+        """Tests warnings collection"""
+        warnings = []
+        pkgs_dict = find_packages(test_data_dir, warnings=warnings)
+
+        warnings.sort()
+        test_expected_warnings.sort()
+        for w, we in zip(warnings, test_expected_warnings):
+            assert w == we, 'Warning message "%s" != "%s"' % (w,we)
```

Comment:
The assertion could get really weird as soon as the two lists are not aligned anymore. I would recommend to simply compare two sets with a single assert:

    assert set(warnings) == set(test_expected_warnings), 'Unexpected difference in warnings: %s != %s' % (warnings, test_expected_warnings)